src/server/editorServices.ts

```diff
@@ -1341,20 +1348,19 @@ namespace ts.server {
                 const parsedCommandLine = ts.parseJsonConfigFileContent(rawConfig.config, this.host, dirPath, /*existingOptions*/ {}, configFilename);
                 Debug.assert(!!parsedCommandLine.fileNames);
 
-                if (parsedCommandLine.errors && (parsedCommandLine.errors.length > 0)) {
-                    return { succeeded: false, errors: parsedCommandLine.errors };
-                }
-                else if (parsedCommandLine.fileNames.length === 0) {
+                if (parsedCommandLine.fileNames.length === 0) {
                     const error = createCompilerDiagnostic(Diagnostics.The_config_file_0_found_doesn_t_contain_any_source_files, configFilename);
-                    return { succeeded: false, errors: [error] };
+                    return { succeeded: false, errors: concatenate(parsedCommandLine.errors, [error]) };

```

Comment:
They are still different though. `succeeded` is true means we can proceed with building projects even if some minor errors exist, but `succeeded` is false means the errors are too serious that we are not going to build anything. Either invalid json and empty project will result in the later case.